src/messages/mod.rs

```diff
@@ -68,6 +67,8 @@ pub struct HopMessageWithBytes {
     content: SignedRoutingMessage,
     /// Serialized Message as received or sent to quic_p2p.
     full_message_bytes: Bytes,
+    /// Crypto hahs of the full message.
+    full_message_crypto_hash: Digest,
```

Comment:
Minor, but do we need to include `crypto` in the name? Can we just assume that "hash" means "crypto hash:?

Reply:
not really, we have Hash derive a lot and the .hash function is not crypto. I'd feel a bit more comfortable keeping it.